packages/components/src/components/SquigglePlayground/index.tsx

```diff
@@ -262,30 +175,21 @@ export const SquigglePlayground: FC<PlaygroundProps> = ({
 
   const [imports, setImports] = useState<JsImports>({});
 
+  const defaultValues: FormFields = {
+    ...schema.getDefault(),
+    ...Object.fromEntries(
+      Object.entries(props).filter(([k, v]) => v !== undefined)
+    ),
+  };
+
   const { register, control } = useForm({
     resolver: yupResolver(schema),
-    defaultValues: {
-      sampleCount: 1000,
-      xyPointLength: 1000,
-      chartHeight: 150,
-      logX,
-      expY,
-      title,
-      minX,
-      maxX,
-      tickFormat,
-      distributionChartActions,
-      showSummary,
-      showEditor,
-      diagramStart: 0,
-      diagramStop: 10,
-      diagramCount: 20,
-    },
-  });
-  const vars = useWatch({
-    control,
+    defaultValues,
   });
 
+  // react-hook-form types the result as Partial, but the result doesn't seem to be a Partial, so this should be ok
```

Comment:
Nice job noticing this. I hope it stays true. Was a bit confused by this returning partials

Reply:
Yeah, I spent quite a bit of time looking through react-hook-form issues and looking at its source code to make sure we're safe here.

I think their types are messy because of defaultValues being optional, and they don't pass the shape of provided defaultValues into the form's type. My impression is that they'll fix this in their next release (8.0), or at least improve something in that area.